control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -105,11 +105,10 @@ func (r *Reconciler) reconcileKind(ctx context.Context, broker *eventing.Broker)
 
 	logger.Debug("config resolved", zap.Any("config", topicConfig))
 
-	securityOption, secret, err := security.NewOptionFromSecret(ctx, &security.MTConfigMapSecretLocator{ConfigMap: brokerConfig}, r.SecretProviderFunc())
+	secret, err := security.Secret(ctx, &security.MTConfigMapSecretLocator{ConfigMap: brokerConfig}, r.SecretProviderFunc())
 	if err != nil {
 		return fmt.Errorf("failed to create security (auth) option: %w", err)
```

Comment:
Nit: should we change this error message?
```suggestion
		return fmt.Errorf("failed to get secret: %w", err)
```

Reply:
sounds good